src/components/character-count/character-count.test.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @jest-environment ./lib/puppeteer/environment.js
+ */
+/* eslint-env jest */
+
+const configPaths = require('../../../config/paths.json')
+const PORT = configPaths.ports.test
+
+let browser
+let page
+let baseUrl = 'http://localhost:' + PORT
+
+beforeEach(async () => {
+  browser = global.__BROWSER__
+  page = await browser.newPage()
+})
+
+afterEach(async () => {
+  await page.close()
+})
+
+describe('Character count', () => {
+  describe('when JavaScript is unavailable or fails', () => {
+    it('shows the static message', async () => {
+      await page.setJavaScriptEnabled(false)
+
+      await page.goto(baseUrl + '/components/character-count/preview', { waitUntil: 'load' })
+      const message = await page.evaluate(() => document.body.querySelector('.govuk-character-count__message').innerHTML.trim())
+
+      expect(message).toEqual('You can enter up to 10 characters')
+    })
+  })
+  describe('when JavaScript is available', () => {
+    it('shows the dynamic message', async () => {
+      await page.goto(baseUrl + '/components/character-count/preview', { waitUntil: 'load' })
+
+      const message = await page.evaluate(() => document.body.querySelector('.govuk-character-count__message').innerHTML)
+
+      expect(message).toEqual('You have 10 characters remaining')
+    })
+
+    it('counts down while typing and within limit', async () => {
+      await page.goto(baseUrl + '/components/character-count/preview', { waitUntil: 'load' })
+
+      // Press 1 character
+      await page.focus('.js-character-count')
+      await page.keyboard.press('A')
+      const message = await page.evaluate(() => document.body.querySelector('.govuk-character-count__message').innerHTML)
+
+      expect(message).toEqual('You have 9 characters remaining')
+    })
+
+    it('counts up while typing and the limit is exceeded', async () => {
+      await page.goto(baseUrl + '/components/character-count/preview', { waitUntil: 'load' })
+
+      // Press 11 characters
+      await page.focus('.js-character-count')
+      for (let i = 0; i < 11; i++) {
```

Comment:
Not blocking: would `page.keyboard.type('A'.repeat(11))` have the same effect?

Reply:
Repeat doesn't seem to be working, unfortunately, so I'll have to leave the form.